libsigopt/compute/domain.py

```diff
@@ -314,13 +325,60 @@ def generate_random_points_near_point(self, num_points, point, std_dev, on_const
     return normal_draws
 
 
+class IntervalDomainComponent(TypedDict):
+  var_type: Literal["int", "double"]
+  elements: tuple[float, float]
+
+
+class GridDomainComponent(TypedDict):
```

Comment:
can you rename this to `SequentialDomainComponent`, we don't treat "categorical" as grid values so `grid` could be misleading